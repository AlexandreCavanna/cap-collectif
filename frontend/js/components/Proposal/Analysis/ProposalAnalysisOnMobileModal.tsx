import React from 'react'
import { Modal } from 'react-bootstrap'
import { FormattedMessage } from 'react-intl'

import styled from 'styled-components'
import CloseButton from '~/components/Form/CloseButton'
import colors from '~/utils/colors'

type Props = {
  show: boolean
  onClose: () => void
}
const Body = styled.div`
  font-size: 15px;
  color: ${colors.darkGray};
`
export const ProposalAnalysisOnMobileModal = ({ show, onClose }: Props) => (
  <Modal show={show} onHide={onClose} aria-labelledby="proposal-analysis-modal-mobile">
    <Modal.Header closeButton>
      <Modal.Title id="proposal-analysis-modal-mobile">
        <FormattedMessage id="analyzing.proposal" />
      </Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <Body>
        <FormattedMessage id="mobile.analysis.unavailable" />
      </Body>
    </Modal.Body>
    <Modal.Footer>
      <CloseButton onClose={onClose} label="global.close" />
    </Modal.Footer>
  </Modal>
)
export default ProposalAnalysisOnMobileModal
