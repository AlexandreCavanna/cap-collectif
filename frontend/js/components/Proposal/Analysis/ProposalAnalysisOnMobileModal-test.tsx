/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { ProposalAnalysisOnMobileModal } from './ProposalAnalysisOnMobileModal'

describe('<ProposalAnalysisOnMobileModal />', () => {
  it('renders correctly', () => {
    const props = {
      show: true,
      onClose: jest.fn(),
    }
    const wrapper = shallow(<ProposalAnalysisOnMobileModal {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
