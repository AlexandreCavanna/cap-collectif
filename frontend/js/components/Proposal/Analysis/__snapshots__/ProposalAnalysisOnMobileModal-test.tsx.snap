// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalAnalysisOnMobileModal /> renders correctly 1`] = `
<Modal
  animation={true}
  aria-labelledby="proposal-analysis-modal-mobile"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[MockFunction]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
      id="proposal-analysis-modal-mobile"
    >
      <FormattedMessage
        id="analyzing.proposal"
        values={Object {}}
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <ProposalAnalysisOnMobileModal__Body>
      <FormattedMessage
        id="mobile.analysis.unavailable"
        values={Object {}}
      />
    </ProposalAnalysisOnMobileModal__Body>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <CloseButton
      label="global.close"
      onClose={[MockFunction]}
    />
  </ModalFooter>
</Modal>
`;
